File: apihub-gw/apihub-gw-api/src/main/java/com/github/thomasandre84/apihub/gw/api/resource/v1/FileProviderResource.java
package com.github.thomasandre84.apihub.gw.api.resource.v1;

import com.github.thomasandre84.apihub.common.rest.FileProviderResourceIf;
import com.github.thomasandre84.apihub.gw.core.service.FileProviderService;
import jakarta.ws.rs.Path;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

import java.io.File;
import java.io.InputStream;
import java.util.List;

@Slf4j
@Path(FileProviderResourceIf.BASE_URL)
@RequiredArgsConstructor
public class FileProviderResource implements FileProviderResourceIf {

    private final FileProviderService fileProviderService;
    @Override
    public List<String> listFiles() {
        return fileProviderService.listFiles();
    }

    @Override
    public File getFile(String fileName) {
        return fileProviderService.getFile(fileName);
    }

    @Override
    public InputStream getFileStream(String fileName) {
        return fileProviderService.getFileStream(fileName);
    }
}

File: apihub-mock/apihub-mock-provider/src/main/kotlin/com/github/thomasandre84/FileProviderResource.kt
package com.github.thomasandre84

import com.github.thomasandre84.apihub.common.rest.FileProviderResourceIf
import jakarta.ws.rs.Path
import jakarta.ws.rs.PathParam
import jakarta.ws.rs.core.StreamingOutput
import java.io.*
import java.nio.file.Files

@Path(FileProviderResourceIf.BASE_URL)
class FileProviderResource: FileProviderResourceIf {

    val folder = java.nio.file.Path.of("/tmp/test/");


    override fun listFiles(): List<String> {
        val files = Files.list(folder)
        return files.map { it.toFile().name }.toList()
    }



    override fun getFile(@PathParam("fileName") fileName: String): File {
        return File(folder.toFile(), fileName)
    }


    override fun getFileStream(@PathParam("fileName") fileName: String): InputStream {
        val file = File(folder.toFile(), fileName)
        val input = BufferedInputStream(FileInputStream(file))
        //val out = BufferedOutputStream(FileOutputStream(file));

        return input;
    }

}